fake-koji/src/main/java/org/fakekoji/jobmanager/OTool.java

```diff
@@ -0,0 +1,39 @@
+package org.fakekoji.jobmanager;
+
+import org.fakekoji.jobmanager.model.JobUpdateResults;
+import org.fakekoji.jobmanager.model.Project;
+import org.fakekoji.storage.StorageException;
+
+import java.util.List;
+
+public class OTool {
+
+    private final JobUpdater jobUpdater;
+
+    public OTool(final JobUpdater jobUpdater) {
+        this.jobUpdater = jobUpdater;
+    }
+
+    /**
+     * Regenerate all jobs. If job is mssing, is (re)created.
```

Comment:
that comment do not sound valid

Reply:
it's your comment

Reply:
and so suddenly it gaves sense. but the refactoring itsel is unclear.  why special class just for this method?

Reply:
This method was duplicated in JDKProjectManager and JDKTestProjectManager

Reply:
I gueess with all but type. So you solved it. So why it is not in common ancestor of  JDKProjectManager and JDKTestProjectManager? If there is none, amybe there can be? If there can not  be, then the rename to SuperHyperAllPRojectsRegenerator will do.
Maybe better, to move  this - by the new style of varisou appis - to regenerate api. But afaik it shodl not be part of this changeset.  What you did with the duplicated code is good. If it can not be moved to common ancestor of  JDKProjectManager and JDKTestProjectManager, then just rename it. And AllProjectsRegenerator really amy be enough. TY!